### ISSUE TITLE
[Ecosystem] DeepInverse

### ISSUE THREAD

**Issue by Andrewwango** [pytorch-fdn/ecosystem#43]
### Contact emails

andrew.wang@ed.ac.uk

### Project summary

DeepInverse: a PyTorch library for imaging with deep learning

### Project description

DeepInverse is an open-source PyTorch-based library for imaging inverse problems. DeepInverse implements all steps for image reconstruction, including efficient forward operators, defining and solving variational problems and designing and training advanced neural networks, for a wide set of domains (medical imaging, astronomical imaging, remote sensing, computational photography, compressed sensing and more).

Imaging is a foundational technology that has driven much of science and engineering in the past century. Deep neural networks have become ubiquitous in various imaging inverse problems. Despite the ever-increasing research effort, most learning-based algorithms are built from scratch, are hard to generalize beyond their specific training setting, and the reported results are often hard to reproduce. DeepInverse overcomes these limitations by providing a modular unified framework, leveraging the popular PyTorch deep learning library. For our audience of researchers (experts in optimization, deep learning etc.), practitioners (biologists, physicists etc.) and imaging software engineers, DeepInverse is:

- Accelerating research by enabling efficient testing, deployment and transfer of new ideas across imaging domains;
- Enlarging the adoption of deep learning in inverse problems by lowering the entrance bar to new users;
- Enhancing research reproducibility via a common modular framework of problems and algorithms.

### Are there any other projects in the PyTorch Ecosystem similar to yours?  If, yes, what are they?

No

### Project repo URL

https://github.com/deepinv/deepinv

### Project license

BSD-3-Clause license

### GitHub handles of the project maintainer(s)

tachella, andrewwango, samuro95, mh-nguyen712, jscanvic

### Is there a corporate or academic entity backing this project?  If so, please provide the name and URL of the entity.

No

### Website URL

https://deepinv.github.io/

### Documentation

Yes - see:

- Quickstart: https://deepinv.github.io/deepinv/quickstart.html
- Examples: https://deepinv.github.io/deepinv/auto_examples/index.html
- User Guide: https://deepinv.github.io/deepinv/user_guide.html

The documentation recently won [a national prize](https://www.ouvrirlascience.fr/presentation-of-the-2024-open-science-free-research-software-awards/).

### How do you build and test the project today (continuous integration)?  Please describe.

Our CI [workflows](https://github.com/deepinv/deepinv/tree/main/.github/workflows) include:
- Linting: `black`, `ruff`
- Testing: `pytest` https://github.com/deepinv/deepinv/tree/main/deepinv/tests
- Docs: `sphinx` and `sphinx-gallery`

### Version of PyTorch

`>=2.2.0`

### Components of PyTorch

All the differentiable components of our imaging pipelines, such as neural network solver models and physics forward operator models inherit from `nn.Module`, with optional buffers. Our library's `Trainer` makes use of this differentiability to train these networks using torch optimizers and our custom suite of loss functions. All our data (i.e. images and imaging measurements) are described as torch tensors. We also make heavy use of GPU for training and inference.

### How long do you expect to maintain the project?

>5 years.

**Comment by tachella** [comment]
👋@lantiga 

Many thanks for the feedback and spotting the broken link.

We've fixed the quickstart link, let us know if something else is needed!


> Thank you @Andrewwango for the submission, great project (I was a medical imaging person in my previous life so I'm happy something like this exists).
> 
> Nit: the quickstart is there, but the link to the quickstart from the README is broken (points to https://deepinv.github.io/deepinv/auto_examples/basics/quickstart.html which 404's).
> 
> **Functional Requirements:**
> 
> ✅ 1. **PyTorch Support:** Project **must** demonstrate benefit to the pytorch community and be validated with the latest versions of PyTorch.
> ✅ 2. **Working CI:** Projects **must** have a demonstrable working CI workflow for all declared architectures
> ✅ 3. **Governance**: Projects **must** have documented technical governance defined and implemented and it must be represented in their Github or Gitlab repository. Example: Review the LF’s Minimum Viable Governance framework.
> ✅ 4. **User experience**: The value proposition **should** be made clear for the project and documented in the README of the project repository.
> 🟡 5. **Quick start:** Project **should** have a working and demonstrable quick start.
> ✅ 6. **Documentation**: Projects **must** have clear and comprehensive documentation.
> ✅ 7. **Permissive licensing:** Users **must** be able to utilize ecosystem projects without licensing concerns. e.g. BSD-3, Apache-2 and MIT licenses
> ✅ 8. **Functional Testing:** Users **should** have developed tests submitted as a part of their project, and be confident that the project will work well with the latest releases of PyTorch.
> ✅ 9. **Packages:** Projects **need** to have support for binary installation options (pip/Conda).
> ✅ 10. **Ongoing maintenance:** Project maintainers **must** be committed to supporting and maintaining their projects for the next 12 months at a minimum. Projects must commit to updating their projects to the latest two releases of PyTorch (e.g. versions 2.6, 2.5.1).
> 
> **Measurable Requirements:**
> 
> ✅ 1. **Core Maintainers:** The project **must** have a minimum of 2 core maintainers
> ✅ 2. **Contributors:** The project **should** have a minimum of 5 total contributors active in the last 90 days.
> ✅ 3. **Commits:** The project **must** have 30 commits in the last 90 days.
> ✅ 4. **Stars:** The project **must** have a minimum of 200 stars on github
> 
> **Github Layout Requirements:**
> 
> ✅ 1. **[LICENSE.md](http://license.md/)** at the root of the repository specifying the terms and conditions for using, distributing, and modifying the software. In addition, you should provide information on the license of any third-party code included in the project.
> ✅ 2. **[README.md](http://readme.md/)** welcomes new community members to the project and explains why the project is useful and how to get started. Should include information about release methodology, cadence, and criteria in the README.
> ✅ 3. **[CONTRIBUTING.md](http://contributing.md/)** explains how to contribute to the project. The file explains the types of contributions needed and how the development process works.
> ✅ 4. **CODEOWNERS** that define individuals or teams responsible for code in a repository; document current project owners and Retired committers.
> ✅ 5. **CODE_OF_CONDUCT.md** sets the ground rules for participants’ behavior and helps facilitate a friendly, welcoming environment. By default, projects should leverage the [[Linux](https://lfprojects.org/policies/code-of-conduct/)](https://lfprojects.org/policies/code-of-conduct/) [[Foundation](https://lfprojects.org/policies/code-of-conduct/)](https://lfprojects.org/policies/code-of-conduct/) [[Code](https://lfprojects.org/policies/code-of-conduct/)](https://lfprojects.org/policies/code-[of](https://lfprojects.org/policies/code-of-conduct/)-conduct/) [of](https://lfprojects.org/policies/code-of-conduct/) [[Conduct](https://lfprojects.org/policies/code-of-conduct/)](https://lfprojects.org/policies/code-of-conduct/) unless an alternate Code of Conduct is approved prior.

**Comment by michellearoth** [comment]
Hi @hogepodge can you please do the 6 month review for DeepInverse?

**Comment by michellearoth** [comment]
@hogepodge reminder for the 6 month review please.

**Comment by hogepodge** [comment]
DeepInverse is actively maintained and continues to meet the requirements for landscape inclusion.

**Comment by thesteve0** [comment]
Agreed
